### PR TITLE
[Sprint 1]Fix_feature_login_logic

### PR DESCRIPTION
**Pull request by NguyenTienVinh**

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/login/LoginModule.java [hunk 1]
```diff
@@ -28,8 +29,9 @@ public LoginModule(@NonNull Activity activity) {
     @ActivityScope
     @Provides
     public LoginContract.ViewModel provideViewModel(Context context,
-            LoginContract.Presenter presenter, Navigator navigator) {
-        return new LoginViewModel(context, presenter, navigator);
+            LoginContract.Presenter presenter, Navigator navigator,
+            DialogManagerImpl dialogManager) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginModule.java]: change to ``DialogManager``

File: app/src/main/java/com/framgia/wsm/screen/login/LoginModule.java [hunk 2]
```diff
@@ -56,4 +58,10 @@ public Navigator provideNavigator() {
     public Validator provideValidator() {
         return new Validator(mActivity.getApplicationContext(), LoginViewModel.class);
     }
+
+    @ActivityScope
+    @Provides
+    public DialogManagerImpl provideDialogManager() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginModule.java]: like above

File: app/src/main/java/com/framgia/wsm/screen/login/LoginViewModel.java [hunk 3]
```diff
@@ -54,8 +59,9 @@ public void onStop() {
     }
 
     @Override
-    public void onLoginError(BaseException throwable) {
-        //TODO: Show Dialog login error!
+    public void onLoginError(BaseException e) {
+        mDialogManager.dismissProgressDialog();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginViewModel.java]: Please do the same action when loginSuccess

File: app/src/main/java/com/framgia/wsm/screen/login/LoginViewModel.java [hunk 4]
```diff
@@ -54,13 +58,16 @@ public void onStop() {
...
+        mNavigator.showToast(e.getMessage());
     }
 
     @Override
     public void onLoginSuccess() {
+        mDialogManager.dismissProgressDialog();
         mNavigator.startActivity(MainActivity.class);
+        mNavigator.showToast(R.string.login_success);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginViewModel.java]: Please remove Toast

File: app/src/main/java/com/framgia/wsm/data/source/local/UserLocalDataSource.java [hunk 5]
```diff
@@ -7,6 +7,7 @@
 import com.framgia.wsm.data.source.UserDataSource;
 import com.framgia.wsm.data.source.local.sharedprf.SharedPrefsApi;
 import com.framgia.wsm.data.source.local.sharedprf.SharedPrefsKey;
+import com.framgia.wsm.data.source.local.sqlite.UserDbHelper;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/source/local/UserLocalDataSource.java]: Please remove package ``sqlite`` 
move all file in this package to package outside ``local``

File: app/src/main/java/com/framgia/wsm/screen/login/LoginModule.java [hunk 6]
```diff
@@ -41,8 +46,22 @@ public LoginModule(@NonNull Activity activity) {
...
+    @ActivityScope
+    @Provides
+    public UserLocalDataSource provideUserLocalDataSource(Context context,
+            SharedPrefsApi sharedPrefsApi) {
+        return new UserLocalDataSource(context, sharedPrefsApi);
+    }
+
+    @ActivityScope
+    @Provides
+    public SharedPrefsApi provideSharedPrefsApi() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginModule.java]: Please remove this line
SharedPrefsApi was declared in ``@AppScope`` u can use without provide in here

Comment (NguyenTienVinh) [inline app/src/main/java/com/framgia/wsm/screen/login/LoginModule.java]: Không pỏ được đâu anh ơi, nó văng lỗi liền

**Pull request merged**